util/testutil/integration/sandbox.go

```diff
@@ -322,7 +337,15 @@ func CheckFeatureCompat(t *testing.T, sb Sandbox, reason ...string) {
 	var ereasons []string
 	for _, r := range reason {
 		if _, ok := features[r]; ok {
-			if b.isUnsupportedFeature(r) {
+			envFeatureName := "BUILDKIT_TEST_FEATURE_" + strings.ToUpper(r)
```

Comment:
I agree. The code is much simpler with single-env, and this `envVars` hack in `hack/test` is not needed.